src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java

```diff
@@ -522,4 +523,53 @@ public void rolesArray() throws Exception {
         Assert.assertTrue(creds.getBackendRoles().contains("3rd"));
     }
 
+    @Test
```

Comment:
For your consideration breaking out the happy path and failure path cases into separate test methods makes it clearer what is being verify.

It might be useful to refactor the call pattern of the tests with a private method with a signature like:

```java
/** extracts a default user credential from a request header */ 
private AuthCredentials extractCredentialsFromJwtHeader(
        final Settings.Builder settingsBuilder,
        final JwtBuilder jwtBuilder) {...}
```

That could make a test look more like 
```java
public void testRequiredAudienceMatches() {
    final AuthCredentials credentials = extractCredentialsFromJwtHeader(
        Settings.builder().put("required_audience", "test_audience"),
        Jwts.builder().setAudience("test_audience"));

    Assert.assertEquals("Leonard McCoy", creds.getUsername());
}
```

Reply:
i took the liberty of creating a patch to take care of this, however i do not have write access to @jochenkressin's fork (as i'm neither him nor a maintainer here), so i'll just drop it here as a diff. feel free to incorporate (or adapt it):
```patch
diff --git a/src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java b/src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java
index 9eeb4b7f..9769dee6 100644
--- a/src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java
+++ b/src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java
@@ -15,6 +15,7 @@
 
 package com.amazon.dlic.auth.http.jwt;
 
+import io.jsonwebtoken.JwtBuilder;
 import io.jsonwebtoken.JwtParser;
 import io.jsonwebtoken.Jwts;
 import io.jsonwebtoken.SignatureAlgorithm;
@@ -524,52 +525,48 @@ public class HTTPJwtAuthenticatorTest {
     }
 
     @Test
-    public void testRequiredAudience() {
-        Settings settings = Settings.builder().put("signing_key", BaseEncoding.base64().encode(secretKey)).put("required_audience", "test_audience")
-                .build();
-
-        String jwsToken = Jwts.builder().setSubject("Leonard McCoy").setAudience("test_audience")
-                .signWith(Keys.hmacShaKeyFor(secretKey), SignatureAlgorithm.HS512).compact();
-
-        HTTPJwtAuthenticator jwtAuth = new HTTPJwtAuthenticator(settings, null);
-        Map<String, String> headers = new HashMap<>();
-        headers.put("Authorization", jwsToken);
-
-        AuthCredentials creds = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<>()), null);
-        Assert.assertNotNull(creds);
-        Assert.assertEquals("Leonard McCoy", creds.getUsername());
-
-        jwsToken = Jwts.builder().setSubject("Leonard McCoy").setAudience("wrong_audience")
-                .signWith(Keys.hmacShaKeyFor(secretKey), SignatureAlgorithm.HS512).compact();
-        headers.put("Authorization", jwsToken);
-
-        creds = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<>()), null);
-
-        Assert.assertNull(creds);
+    public void testRequiredAudienceWithCorrectAudience() {
+        final AuthCredentials credentials = extractCredentialsFromJwtHeader(
+                Settings.builder().put("signing_key", BaseEncoding.base64().encode(secretKey)).put("required_audience", "test_audience"),
+                Jwts.builder().setSubject("Leonard McCoy").setAudience("test_audience").signWith(Keys.hmacShaKeyFor(secretKey), SignatureAlgorithm.HS512));
+        Assert.assertNotNull(credentials);
+        Assert.assertEquals("Leonard McCoy", credentials.getUsername());
     }
 
     @Test
-    public void testRequiredIssuer() {
-        Settings settings = Settings.builder().put("signing_key", BaseEncoding.base64().encode(secretKey)).put("required_issuer", "test_issuer")
-                .build();
-
-        String jwsToken = Jwts.builder().setSubject("Leonard McCoy").setIssuer("test_issuer")
-                .signWith(Keys.hmacShaKeyFor(secretKey), SignatureAlgorithm.HS512).compact();
-
-        HTTPJwtAuthenticator jwtAuth = new HTTPJwtAuthenticator(settings, null);
-        Map<String, String> headers = new HashMap<>();
-        headers.put("Authorization", jwsToken);
-
-        AuthCredentials creds = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<>()), null);
-        Assert.assertNotNull(creds);
-        Assert.assertEquals("Leonard McCoy", creds.getUsername());
+    public void testRequiredAudienceWithIncorrectAudience() {
+        final AuthCredentials credentials = extractCredentialsFromJwtHeader(
+                Settings.builder().put("signing_key", BaseEncoding.base64().encode(secretKey)).put("required_audience", "test_audience"),
+                Jwts.builder().setSubject("Leonard McCoy").setAudience("wrong_audience").signWith(Keys.hmacShaKeyFor(secretKey), SignatureAlgorithm.HS512));
+        Assert.assertNull(credentials);
+    }
 
-        jwsToken = Jwts.builder().setSubject("Leonard McCoy").setIssuer("wrong_issuer")
-                .signWith(Keys.hmacShaKeyFor(secretKey), SignatureAlgorithm.HS512).compact();
-        headers.put("Authorization", jwsToken);
+    @Test
+    public void testRequiredIssuerWithCorrectAudience() {
+        final AuthCredentials credentials = extractCredentialsFromJwtHeader(
+                Settings.builder().put("signing_key", BaseEncoding.base64().encode(secretKey)).put("required_issuer", "test_issuer"),
+                Jwts.builder().setSubject("Leonard McCoy").setIssuer("test_issuer").signWith(Keys.hmacShaKeyFor(secretKey), SignatureAlgorithm.HS512));
+        Assert.assertNotNull(credentials);
+        Assert.assertEquals("Leonard McCoy", credentials.getUsername());
+    }
 
-        creds = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<>()), null);
+    @Test
+    public void testRequiredIssuerWithIncorrectAudience() {
+        final AuthCredentials credentials = extractCredentialsFromJwtHeader(
+                Settings.builder().put("signing_key", BaseEncoding.base64().encode(secretKey)).put("required_issuer", "test_issuer"),
+                Jwts.builder().setSubject("Leonard McCoy").setIssuer("wrong_issuer").signWith(Keys.hmacShaKeyFor(secretKey), SignatureAlgorithm.HS512));
+        Assert.assertNull(credentials);
+    }
 
-        Assert.assertNull(creds);
+    /** extracts a default user credential from a request header */
+    private AuthCredentials extractCredentialsFromJwtHeader(
+            final Settings.Builder settingsBuilder,
+            final JwtBuilder jwtBuilder) {
+        final Settings settings = settingsBuilder.build();
+        final String jwsToken = jwtBuilder.compact();
+        final HTTPJwtAuthenticator jwtAuth = new HTTPJwtAuthenticator(settings, null);
+        final Map<String, String> headers = Map.of("Authorization", jwsToken);
+        final AuthCredentials credentials = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<>()), null);
+        return credentials;
     }
 }

```

(i hope that with this adaptation the PR then gets the two approvals and can be merged soon ;))